### PR TITLE
Exclude parsing null values from manifests

### PR DESCRIPTION
**Pull request by rickbijkerk**
null values causes the in memory hashes/operaties map to be reset to null

### CODE REVIEW

File: internal/business/persisted_operations/dir_loader.go [hunk 1]
```diff
@@ -49,6 +50,11 @@ func (d *DirLoader) Load(_ context.Context) (map[string]string, error) {
 				continue
 			}
 
+			// Dont parse null values, as this causes the hash/operations map to be reset to null
+			if isNullValue(contents) {
```
[the remark this answers is not in the document]
Comment (rickbijkerk) [reply internal/business/persisted_operations/dir_loader.go:54]: nice suggestion let me check that

File: internal/business/persisted_operations/dir_loader.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func (d *DirLoader) Load(_ context.Context) (map[string]string, error) {
 		}
 	}
 
-	var result map[string]string
+	var result = map[string]string{}
```
Comment (ldebruijn) [inline internal/business/persisted_operations/dir_loader.go:39]:
```suggestion
	var result = map[string]string
```

File: internal/business/persisted_operations/dir_loader.go [hunk 3]
```diff
@@ -49,11 +49,16 @@ func (d *DirLoader) Load(_ context.Context) (map[string]string, error) {
...
 
-			err = json.Unmarshal(contents, &result)
+			var manifestHashes map[string]string
+
+			err = json.Unmarshal(contents, &manifestHashes)
 			if err != nil {
 				d.log.Warn("error unmarshalling operation file", "filepath", filePath, "err", err)
 				continue
 			}
+			for hash, operation := range manifestHashes {
```
Comment (ldebruijn) [inline internal/business/persisted_operations/dir_loader.go]:
```suggestion
			maps.Copy(result, manifestHashes)
```

File: internal/business/persisted_operations/dir_loader.go (hunk 2 above)
Comment (rickbijkerk) [reply internal/business/persisted_operations/dir_loader.go:39]: go doesnt like that

**Review by ldebruijn — APPROVED** [review]
Great finding!

**Pull request merged**